test/e2e/util.go

```diff
@@ -482,35 +539,49 @@ func RunRC(c *client.Client, name string, ns, image string, replicas int) error
 		return fmt.Errorf("Error creating replication controller: %v", err)
 	}
 
+	// Create a routine to query for the list of pods
+	stop := make(chan struct{})
+	go func(stop <-chan struct{}, n string, ns string, l labels.Selector) {
+		for {
+			select {
+			case <-stop:
+				return
+			default:
+				p, err := c.Pods(ns).List(l, fields.Everything())
+				if err != nil {
+					Logf("Warning: Failed to get pod list: %v", err)
+				} else {
+					podLists.Push(p.Items)
+				}
+				time.Sleep(1 * time.Second)
+			}
+		}
+	}(stop, name, ns, label)
+
 	By(fmt.Sprintf("Making sure all %d replicas exist", replicas))
-	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
-	pods, err := listPods(c, ns, label, fields.Everything())
-	if err != nil {
-		return fmt.Errorf("Error listing pods: %v", err)
-	}
-	current = len(pods.Items)
 	failCount := 5
+	var pods []api.Pod
 	for same < failCount && current < replicas {
-		Logf("Controller %s: Found %d pods out of %d", name, current, replicas)
-		if last < current {
-			same = 0
-		} else if last == current {
-			same++
-		} else if current < last {
-			return fmt.Errorf("Controller %s: Number of submitted pods dropped from %d to %d", name, last, current)
-		}
+		time.Sleep(2 * time.Second)
+		for podLists.Len() > 0 && current < replicas {
```

Comment:
The section checking that the controller is submitting the expected number of pods is using the same queue as the later section checking the states.  Here it is only looking at the total number of pops but it's not really using any more resources since it's a shared queue.  If a watch would work for both I'm open to looking at it, but I don't see the point in using a watch here if I still need the queue later.